storagemarket/lp2pimpl/net.go

```diff
@@ -89,26 +95,30 @@ func NewDealClient(h host.Host, options ...DealClientOption) *DealClient {
 
 // DealProvider listens for incoming deal proposals over libp2p
 type DealProvider struct {
-	host host.Host
-	prov *storagemarket.Provider
+	host     host.Host
+	prov     *storagemarket.Provider
+	fullNode v1api.FullNode
 }
 
-func NewDealProvider(h host.Host, prov *storagemarket.Provider) *DealProvider {
+func NewDealProvider(h host.Host, prov *storagemarket.Provider, fullNodeApi v1api.FullNode) *DealProvider {
 	p := &DealProvider{
-		host: h,
-		prov: prov,
+		host:     h,
+		prov:     prov,
+		fullNode: fullNodeApi,
 	}
 	return p
 }
 
 func (p *DealProvider) Start() {
 	p.host.SetStreamHandler(DealProtocolID, p.handleNewDealStream)
 	p.host.SetStreamHandler(mktssm.AskProtocolID, p.handleNewAskStream)
+	p.host.SetStreamHandler(mktssm.DealStatusProtocolID, p.handleNewDealStatusStream)
```

Comment:
Can you create an issue to remove this handler from `go-fil-markets` then after the integration ?